keps/sig-storage/1682-csi-driver-skip-permission/README.md

```diff
@@ -0,0 +1,236 @@
+)
+
+type CSIDriverSpec struct {
+    // SupportsFSGroup ← new field
+    // Defines if the underlying volume supports changing ownership and 
+    // permission of the volume before being mounted. 
+    // If set to Always, SupportsFSGroup indicates that 
+    // the underlying volume supports volume ownership and permission
+    // changes, and the filesystem will always be modified to match the defined
+    // fsGroup.
+    // If set to Never, then the volume will be mounted without modifying
+    // the volume's ownership or permissions.
+    // Defaults to OnlyRWO, which results in the volume being examined
+    // and the volume ownership and permissions attempting to be updated
+    // if fsGroup is defined and accessModes is RWO.
+    // + optional
+    SupportsFSGroup *SupportsFsGroup
+}
+```
+### Test Plan
+
+A test plan will include the following tests:
+
+* Basic tests including a permutation of the following values:
+  - CSIDriver.Spec.SupportsFSGroup  (`Always`/`Never`/`OnlyRWO`)
+  - PersistentVolumeClaim.Status.AccessModes (`ReadWriteOnly`, `ReadOnlyMany`,`ReadWriteMany`)
+* E2E tests
+
+### Graduation Criteria
+
+* Alpha in 1.19 provided all tests are passing.
+* All functionality is guarded by a new alpha `CSIVolumeSupportFSGroup` feature gate.
+
+* Beta in 1.20 with design validated by at least two customer deployments
+  (non-production), with discussions in SIG-Storage regarding success of
+  deployments. 
+* The `CSIVolumeSupportFSGroup` feature gate will graduate to beta.
+
+
+* GA in 1.21, with E2E tests in place tagged with feature Storage.
+* The `CSIVolumeSupportFSGroup` feature gate will graduate to GA.
+
+[issues]: https://github.com/kubernetes/enhancements/issues/1682
+
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: CSIVolumeSupportFSGroup 
+    - Components depending on the feature gate: No k8s components
+
+* **Does enabling the feature change any default behavior?**
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Also set `rollback-supported` to `true` or `false` in `kep.yaml`.
+  Describe the consequences on existing workloads (e.g. if this is runtime
+  feature, can it break the existing applications?).
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling and disabling feature
+  gates. However, unit tests in each component dealing with managing data created
+  with and without the feature are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
```

Comment:
The section "Feature enablement and rollback" is required for alpha features.  Can you please answer them?

Reply:
These questions have been answered.